File: projects/demo/platform_admin/backend/common/auth/model/admin_config.go
package model

import (
	"time"

	"gorm.io/gorm"
)

// AdminConfig 三级配置模型
type AdminConfig struct {
	ID            int64          `gorm:"primaryKey" json:"id,string"`
	ConfigKey     string         `gorm:"type:varchar(128);not null;uniqueIndex:uk_scope_key" json:"config_key"`
	ConfigValue   string         `gorm:"type:text" json:"config_value"`
	ConfigType    string         `gorm:"type:varchar(32);not null;default:'string'" json:"config_type"`              // string/number/boolean/json
	Scope         string         `gorm:"type:varchar(16);not null;default:'SYSTEM';uniqueIndex:uk_scope_key" json:"scope"` // SYSTEM/TENANT/USER
	ScopeID       int64          `gorm:"not null;default:0;uniqueIndex:uk_scope_key" json:"scope_id,string"`         // SYSTEM=0, TENANT=tenantId, USER=userId
	TenantID      int64          `gorm:"not null;default:0;uniqueIndex:uk_scope_key" json:"tenant_id,string"`        // 所属租户(SYSTEM=0)
	DisplayName   string         `gorm:"type:varchar(128)" json:"display_name"`
	Description   string         `gorm:"type:varchar(512)" json:"description"`
	IsFeatureFlag int            `gorm:"not null;default:0" json:"is_feature_flag"`                                  // 1=功能开关
	Status        int            `gorm:"not null;default:1" json:"status"`
	DeletedAt     gorm.DeletedAt `gorm:"index" json:"-"`
	CreatedAt     *time.Time     `gorm:"autoCreateTime" json:"created_at"`
	UpdatedAt     *time.Time     `gorm:"autoUpdateTime" json:"updated_at"`
}

// TableName 指定表名
func (AdminConfig) TableName() string {
	return "admin_config"
}

// BeforeCreate 创建前钩子，自动生成雪花 ID
func (ac *AdminConfig) BeforeCreate(tx *gorm.DB) error {
	if ac.ID == 0 {
		ac.ID = NextID()
	}
	return nil
}

File: projects/demo/platform_admin/dev-web-admin/src/api/admin-config.ts
/**
 * 三级配置 API（admin_config 表）
 * 后端路由：/api/v1/admin/configs
 */
import request from '@/utils/request'

/** 配置项 */
export interface AdminConfigItem {
  id: string
  config_key: string
  config_value: string
  config_type: string
  scope: string
  scope_id: string
  tenant_id: string
  display_name: string
  description: string
  is_feature_flag: number
  status: number
  created_at: string
  updated_at: string
}

/** 查询参数 */
export interface AdminConfigQuery {
  scope?: string
  key?: string
  is_feature_flag?: number
  page: number
  page_size: number
}

/** 创建参数 */
export interface AdminConfigCreateParams {
  config_key: string
  config_value: string
  config_type?: string
  scope: string
  scope_id: string
  tenant_id: string
  display_name?: string
  description?: string
  is_feature_flag?: number
}

/** 功能开关项 */
export interface FeatureFlagItem {
  key: string
  display_name: string
  enabled: boolean
}

/** 配置列表 */
export async function listAdminConfigs(params: AdminConfigQuery) {
  const res: any = await request.get('/api/v1/admin/configs', { params })
  return res?.data || res || { list: [], total: 0 }
}

/** 创建配置 */
export function createAdminConfig(data: AdminConfigCreateParams) {
  return request.post('/api/v1/admin/configs', data)
}

/** 更新配置 */
export function updateAdminConfig(id: string, data: Partial<AdminConfigCreateParams>) {
  return request.put(`/api/v1/admin/configs/${id}`, data)
}

/** 删除配置 */
export function deleteAdminConfig(id: string) {
  return request.delete(`/api/v1/admin/configs/${id}`)
}

/** 按 key 解析有效值（三级合并） */
export function resolveConfig(key: string) {
  return request.get(`/api/v1/admin/configs/resolve/${key}`)
}

/** 获取功能开关列表 */
export async function getFeatureFlags(): Promise<FeatureFlagItem[]> {
  const res: any = await request.get('/api/v1/admin/configs/feature-flags')
  return res?.data || res || []
}
